### PR TITLE
feat: new debug postage api

### PR DESCRIPTION
**Pull request by AuHau**
Successor of #355

### CODE REVIEW

**Comment by AuHau** [comment]
It bit depends if the Bee's rename PR will be merged or not https://github.com/ethersphere/bee/pull/2371

File: src/bee-debug.ts [hunk 1]
```diff
@@ -298,4 +309,82 @@ export class BeeDebug {
...
+
+  /**
+   * Return details for specific postage batch.
+   *
+   * @param postageBatchId Batch ID
+   *
+   * @see [Bee docs - Keep your data alive / Postage stamps](https://docs.ethswarm.org/docs/access-the-swarm/keep-your-data-alive)
+   * @see [Bee Debug API reference - `GET /stamps/${id}`](https://docs.ethswarm.org/debug-api/#tag/Postage-Stamps/paths/~1stamps~1{id}/get)
+   */
+  async getPostageBatch(postageBatchId: BatchId | string): Promise<PostageBatch> {
```
Comment (nugaon) [inline src/bee-debug.ts:361]:
```suggestion
  async getPostageBatch(postageBatchId: BatchId | string): Promise<DebugPostageBatch> {
```

File: src/bee-debug.ts [hunk 2]
```diff
@@ -298,4 +309,82 @@ export class BeeDebug {
...
+    return stamps.getPostageBatchBuckets(this.url, postageBatchId)
+  }
+
+  /**
+   * Return all postage batches that has the node available.
+   *
+   * @see [Bee docs - Keep your data alive / Postage stamps](https://docs.ethswarm.org/docs/access-the-swarm/keep-your-data-alive)
+   * @see [Bee Debug API reference - `GET /stamps`](https://docs.ethswarm.org/debug-api/#tag/Postage-Stamps/paths/~1stamps/get)
+   */
+  async getAllPostageBatch(): Promise<PostageBatch[]> {
```
Comment (nugaon) [inline src/bee-debug.ts:387]:
```suggestion
  async getAllPostageBatch(): Promise<DebugPostageBatch[]> {
```

**Review by nugaon — CHANGES_REQUESTED** [review]
I commented two type issues, but others seems good.

On local test run I got error:

```
  ● Bee Debug class › PostageBatch › should have both immutable true and false

    Internal Server Error: cannot create batch
```

File: src/bee-debug.ts (hunk 1 above)
Comment (nugaon) [reply src/bee-debug.ts:361]: Don't commit these changes because the source file does not have `DebugPostageBatch` import.

**Review by nugaon — APPROVED** [review]
LGTM!

**Pull request merged**